src/Centreon/Infrastructure/PlatformTopology/PlatformTopologyRepositoryRDB.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Infrastructure\PlatformTopology;
+
+use Centreon\Domain\Entity\EntityCreator;
+use Centreon\Domain\PlatformTopology\Interfaces\PlatformTopologyRepositoryInterface;
+use Centreon\Domain\PlatformTopology\PlatformTopology;
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+
+/**
+ * This class is designed to manage the repository of the platform topology requests
+ *
+ * @package Centreon\Infrastructure\PlatformTopology
+ */
+class PlatformTopologyRepositoryRDB extends AbstractRepositoryDRB implements PlatformTopologyRepositoryInterface
+{
+    /**
+     * PlatformTopologyRepositoryRDB constructor.
+     * @param DatabaseConnection $db
+     */
+    public function __construct(DatabaseConnection $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addPlatformToTopology(PlatformTopology $platformTopology): void
+    {
+        $statement = $this->db->prepare(
+            $this->translateDbName('
+                INSERT INTO `:db`.platform_topology (`address`, `name`, `type`, `parent_id`, `server_id`)
+                VALUES (:address, :name, :type, :parentId, :serverId)
+            ')
+        );
+        $statement->bindValue(':address', $platformTopology->getAddress(), \PDO::PARAM_STR);
+        $statement->bindValue(':name', $platformTopology->getName(), \PDO::PARAM_STR);
+        $statement->bindValue(':type', strtolower($platformTopology->getType()), \PDO::PARAM_STR);
```

Comment:
have you not already strtolower this value ? your method PlatformTopology::setType already strtolower $type

Reply:
indeed... I'll remove this